src/main/java/com/codingame/codemachine/compiler/java/CodinGameJavaCompiler.java

```diff
@@ -1,126 +1,105 @@
-
-            Iterable<? extends JavaFileObject> compilationUnits = fileManager.getJavaFileObjectsFromStrings(files);
-            JavaCompiler.CompilationTask task =
-                compiler.getTask(null, fileManager, diagnosticsCollector, options, null, compilationUnits);
-            boolean success = task.call();
-            for (Diagnostic<? extends JavaFileObject> diagnostic : diagnosticsCollector.getDiagnostics()) {
-                switch (diagnostic.getKind()) {
+    
+            System.out.println("CG> redirect-streams -i \"err\" \"compilation\"");
+            if (!files.isEmpty()) {
+                Iterable<? extends JavaFileObject> compilationUnits = fileManager.getJavaFileObjectsFromStrings(files);
+                JavaCompiler.CompilationTask task =
+                    compiler.getTask(null, fileManager, diagnosticsCollector, options, null, compilationUnits);
+                boolean success = task.call();
+                for (Diagnostic<? extends JavaFileObject> diagnostic : diagnosticsCollector.getDiagnostics()) {
+                    
+                    LineNumberReader reader = new LineNumberReader(new StringReader(diagnostic.getSource().getCharContent(true).toString()));
+                    String line = reader.lines().skip(diagnostic.getLineNumber() - 1).limit(1).findAny().get();
+                    
+                    String type = null;
+                    switch (diagnostic.getKind()) {
                     case ERROR:
+                        type = "ERROR";
+                        break;
                     case WARNING:
-                        CompilationLogDto log = new CompilationLogDto();
-                        log.setFilename(diagnostic.getSource().getName());
-                        log.setLine(diagnostic.getLineNumber());
-                        log.setColumn(diagnostic.getColumnNumber());
-                        log.setMessage(diagnostic.getMessage(null));
-                        log.setKind(diagnostic.getKind() == Kind.ERROR ? CompilationLogKind.ERROR : CompilationLogKind.WARNING);
-                        logs.add(log);
+                    case MANDATORY_WARNING:
+                        type = "WARNING";
                         break;
-                    default:
-                        // nothing
+                    case NOTE:
+                        type = "INFO";
+                        break;
+                    case OTHER:
+                        continue;
+                    }
+                    
+                    System.err.println(String.format("%s:%d: %s: %s\n%s\n%s^", 
+                            diagnostic.getSource().getName(),
+                            diagnostic.getLineNumber(),
+                            diagnostic.getKind().name().toLowerCase(),
+                            diagnostic.getMessage(null),
+                            line,
+                            String.join("", Collections.nCopies((int)diagnostic.getColumnNumber() - 1, " "))
```

Comment:
`%s` will print the string as it is. `%<n>s` will print the string as it is, but if the string has less than `n` characters, the output will be padded on the left.